File: 3080. Mark Elements on Array by Performing Queries.java
class Solution {
    public long[] unmarkedSumArray(int[] nums, int[][] queries) {
        PriorityQueue<Num> heap = new PriorityQueue<>((a,b) -> {
            if (a.val == b.val) return a.index - b.index;
            return a.val - b.val;
        });
        Set<Integer> markedIndices = new HashSet<>();
        long total = 0;
        for (int i = 0; i < nums.length; i++) {
            total += nums[i];
            heap.add(new Num(nums[i], i));
        }
        long[] answer = new long[queries.length];
        for (int i = 0; i < queries.length; i++) {
            int[] q = queries[i];
            int index = q[0];
            int count = q[1];
            if (!markedIndices.contains(index)) {
                total -= nums[index];
                markedIndices.add(index);
            }
            while (count > 0 && !heap.isEmpty()) {
                Num current = heap.poll();
                if (markedIndices.contains(current.index)) continue;
                markedIndices.add(current.index);
                total -= current.val;
                count--;
            }
            answer[i] = total;
        }
        return answer;
    }
}

class Num {
    int val;
    int index;
    Num(int val, int index) {
        this.val = val;
        this.index = index;
    }
}

File: 3080. Mark Elements on Array by Performing Queries.py
class Solution:
    def unmarkedSumArray(self, nums: List[int], queries: List[List[int]]) -> List[int]:
        heap = []
        heapq.heapify(heap)
        marked = set()
        total = 0
        for i in range(len(nums)):
            total += nums[i]
            heapq.heappush(heap, (nums[i], i))
        res = []
        for query in queries:
            index, count = query
            if index not in marked:
                total -= nums[index]
                marked.add(index)
            while heap and count > 0:
                current = heapq.heappop(heap)
                if current[1] in marked:
                    continue
                total -= current[0]
                marked.add(current[1])
                count -= 1
            res.append(total)
        return res
